qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -661,6 +666,90 @@ def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **
         return decomposition
 
 
+class QftSynthesisFull(HighLevelSynthesisPlugin):
+    """Synthesis plugin for QFT gates using all-to-all connectivity.
+
+    This plugin name is :``qft.full`` which can be used as the key on
+    an :class:`~.HLSConfig` object to use this method with :class:`~.HighLevelSynthesis`.
+
+    The plugin supports the following additional options:
+
+    * do_swaps (bool): Whether to include Swap gates at the end of the synthesized
+        circuit. Some implementation of the ``QftGate`` include a layer of Swap gates
+        at the end of the synthesized circuit, which cna in principle be dropped if
+        the ``QftGate`` itself is the last gate in the circuit.
+    * approximation_degree (int): The degree of approximation (0 for no approximation).
+        It is impossible ti implement the QFT approximately by ignoring
+        controlled-phase rotations with the angle is beneath a threshold. This is discussed
+        in more detail in https://arxiv.org/abs/quant-ph/9601018 or
+        https://arxiv.org/abs/quant-ph/0403071.
+    * insert_barriers (bool): If True, barriers are inserted as visualization improvement.
+    * inverse (bool): If True, the inverse Fourier transform is constructed.
+    * name (str): The name of the circuit.
+
+    """
+
+    def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **options):
+        """Run synthesis for the given QftGate."""
+        if not isinstance(high_level_object, QftGate):
+            raise TranspilerError(
+                "The synthesis plugin 'qft.full` only applies to objects of type QftGate."
+            )
+
+        do_swaps = options.get("do_swaps", True)
+        approximation_degree = options.get("approximation_degree", 0)
+        insert_barriers = options.get("insert_barriers", False)
+        inverse = options.get("inverse", False)
+        name = options.get("name", None)
+
+        decomposition = synth_qft_full(
+            num_qubits=high_level_object.num_qubits,
+            do_swaps=do_swaps,
+            approximation_degree=approximation_degree,
+            insert_barriers=insert_barriers,
+            inverse=inverse,
+            name=name,
+        )
+        return decomposition
+
+
+class QftSynthesisLine(HighLevelSynthesisPlugin):
+    """Synthesis plugin for QFT gates using linear connectivity.
+
+    This plugin name is :``qft.line`` which can be used as the key on
+    an :class:`~.HLSConfig` object to use this method with :class:`~.HighLevelSynthesis`.
+
+    The plugin supports the following additional options:
+
+    * do_swaps (bool): whether to include Swap gates at the end of the synthesized
```

Comment:
again, perhaps it's worth to change it to something like "reverse_qubits" ?
also: cna --> can

Reply:
Done!